lib/utilityservice.js

```diff
@@ -588,4 +589,176 @@ UtilityService.prototype.updateMetadata = async function(type, metadata) {
     
         return this.vlocity.utilityservice.setNamespaceToDefault(result);
     }
-};
\ No newline at end of file
+}
+
+UtilityService.prototype.createOrUpdateGitIgnoreFile = async function (jobInfo) {
+
+    var gitIgnorePath = path.join(jobInfo.localRepoPath, '.gitignore');
+
+    if (fs.existsSync(gitIgnorePath)) {
+        return;
+    }
+
+    var ignoreFile = fs.openSync(gitIgnorePath, 'w');
+    fs.writeSync(ignoreFile, "\nVlocityBuild*\nvlocity-temp/\n");
+    fs.closeSync(ignoreFile);
+}
+
+UtilityService.prototype.runGitCheckoutBranch = async function (jobInfo) {
+
+    if(!jobInfo.enableFullGitSupport) {
+        jobInfo.hasError = true;
+        jobInfo.errors.push('Please set enableFullGitSupport to true in job file settings');
+        VlocityUtils.error('Please set enableFullGitSupport to true in job file settings');
+        return;
+    }
+
+    if (!jobInfo.currentBranch) {
+        jobInfo.hasError = true;
+        jobInfo.errors.push('branch name not entered');
+        VlocityUtils.error('branch name not entered');
+        return;
+    }
+
+    if (!jobInfo.localRepoPath) {
+        jobInfo.localRepoPath = jobInfo.projectPath;
+    }
+
+    await simpleGit(jobInfo.localRepoPath)
+        .checkoutBranch(jobInfo.currentBranch, 'master');
+
+}
+
+UtilityService.prototype.runGitInit = async function (jobInfo) {
+    var self = this;
+
+    if(!jobInfo.enableFullGitSupport) {
+        jobInfo.hasError = true;
+        jobInfo.errors.push('Please set enableFullGitSupport to true in job file settings');
+        VlocityUtils.error('Please set enableFullGitSupport to true in job file settings');
+        return;
+    }
+
+    let promise = await new Promise((resolve) => {
```

Comment:
This seems like ti should not need a Promise here since it is Async. Anything that is a promise can be changed to use Async. Only things that only support a Callback method cannot be made async without wrapping in a Promise.